src/Tree.js

```diff
@@ -52,38 +23,70 @@ export default class Tree extends Component {
         renderNodeContent: RenderNodeContent
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            flatTree: this._buildTree(props.nodes, [])
+        };
+
+        this._renderFlatNode = this._renderFlatNode.bind(this);
+    }
+
+    _buildTree(node, tree, parentId) {
```

Comment:
parameter names are misleading:
* `node` is actually tree
* `tree` is actually list of nodes

Reply:
thx, I'm going to move this code to separated entity and refactor it in this pr